Pegasus/include/Geometry.hpp

```diff
@@ -641,79 +1079,121 @@ inline double CalculatePenetration<Sphere, Box>(SimpleShape const* a, SimpleShap
+    auto cache = static_cast<IntersectionCache<Box, Sphere>*>(cacheBase);
     Initialize<Sphere, Box>(b, a, &cache->sbCache);
 }
 
+/** Box, Sphere CalculateIntersection specialization */
 template <>
-inline bool Overlap<Box, Sphere>(SimpleShape const* a, SimpleShape const* b, CacheBase* cacheBase)
+inline bool CalculateIntersection<Box, Sphere>(SimpleShape const* a, SimpleShape const* b, IntersectionCacheBase* cacheBase)
 {
-    auto cache = static_cast<Cache<Box, Sphere>*>(cacheBase);
-    return Overlap<Sphere, Box>(b, a, &cache->sbCache);
+    auto cache = static_cast<IntersectionCache<Box, Sphere>*>(cacheBase);
+    return CalculateIntersection<Sphere, Box>(b, a, &cache->sbCache);
 }
 
+/** Box, Sphere CalculateContactNormal specialization */
 template <>
-inline glm::dvec3 CalculateContactNormal<Box, Sphere>(SimpleShape const* a, SimpleShape const* b, CacheBase* cacheBase)
+inline glm::dvec3 CalculateContactNormal<Box, Sphere>(SimpleShape const* a, SimpleShape const* b, IntersectionCacheBase* cacheBase)
 {
-    auto cache = static_cast<Cache<Box, Sphere>*>(cacheBase);
+    auto cache = static_cast<IntersectionCache<Box, Sphere>*>(cacheBase);
     CalculateContactNormal<Sphere, Box>(b, a, &cache->sbCache);
 
     return cache->sbCache.sphereContactNormal;
 }
 
+/** Box, Sphere CalculatePenetration specialization */
 template <>
-inline double CalculatePenetration<Box, Sphere>(SimpleShape const* a, SimpleShape const* b, CacheBase* cacheBase)
+inline double CalculatePenetration<Box, Sphere>(SimpleShape const* a, SimpleShape const* b, IntersectionCacheBase* cacheBase)
 {
-    auto cache = static_cast<Cache<Box, Sphere>*>(cacheBase);
+    auto cache = static_cast<IntersectionCache<Box, Sphere>*>(cacheBase);
     return CalculatePenetration<Sphere, Box>(b, a, &cache->sbCache);
 }
 
-// Box, Box
+/** Box, Box Initialize specialization */
 template <>
-inline void Initialize<Box, Box>(SimpleShape const* a, SimpleShape const* b, CacheBase* cacheBase)
+inline void Initialize<Box, Box>(SimpleShape const* a, SimpleShape const* b, IntersectionCacheBase* cacheBase)
 {
-    auto aBox = static_cast<Box const *>(a);
-    auto bBox = static_cast<Box const *>(b);
-    auto cache = static_cast<Cache<Box, Box>*>(cacheBase);
+    auto aBox = static_cast<Box const*>(a);
+    auto bBox = static_cast<Box const*>(b);
+    auto cache = static_cast<IntersectionCache<Box, Box>*>(cacheBase);
 
-    cache->aMassCenter = aBox->GetCenterOfMass();
-    aBox->GetAxes(cache->aBoxAxes[0], cache->aBoxAxes[1], cache->aBoxAxes[2]);
-    cache->aBoxAxes = {cache->aBoxAxes[0], cache->aBoxAxes[1], cache->aBoxAxes[2],
-        cache->aBoxAxes[0] * -1.0, cache->aBoxAxes[1] * -1.0, cache->aBoxAxes[2] * -1.0};
-    CalculateBoxVertices(cache->aBoxAxes[0], cache->aBoxAxes[1], cache->aBoxAxes[2], cache->aBoxVertices.begin());
+    cache->aMassCenter = aBox->centerOfMass;
+    cache->aBoxAxes = { aBox->iAxis, aBox->jAxis, aBox->kAxis };
+    cache->aBoxAxes = {
```

Comment:
could lines 1191 and 1192 be combined? Feels like you're rewriting what you just assigned